src/Common/Validation/RegularExpressionCollectionAttribute.cs

```diff
@@ -0,0 +1,31 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace Passwordless.Common.Validation;
+
+[AttributeUsage(AttributeTargets.Property)]
+public class RegularExpressionCollectionAttribute(string pattern) : RegularExpressionAttribute(pattern)
+{
+    public override bool IsValid(object? value)
+    {
+        if (value == null)
+        {
+            return true;
+        }
+
+        if (value is not IEnumerable<string> collection)
+        {
+            throw new InvalidOperationException("This attribute can only be applied to collections");
```

Comment:
Nit: dots are customary at the end of exception messages

```suggestion
            throw new InvalidOperationException("This attribute can only be applied to collections.");
```